### PR TITLE
Expose GUI labels in RPC as comments

### PR DESCRIPTION
**Pull request by luke-jr**

### CODE REVIEW

**Comment by laanwj** [comment]
I think then we should simply call it 'label', or 'address label'. Any API replacing the account system would use that for consistency with the GUI.

**Comment by laanwj** [comment]
Needs rebase. And I still think the string should have the same name as in the GUI, e.g. 'label', not think up a new term 'comment' that will just cause confusion.

**Comment by Diapolo** [comment]
+1 for a consistent term between GUI and core.

**Comment by laanwj** [comment]
Need rebase(and comment addressed, probably)

**Comment by jgarzik** [comment]
concept ACK, once feedback is addressed

**Comment by jgarzik** [comment]
Ping @luke-jr

**Comment by dcousens** [comment]
concept ACK,  will review on rebase

**Comment by luke-jr** [comment]
Rebased and renamed to "label" (although I think we'll regret that).

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -1182,6 +1182,8 @@ UniValue ListReceived(const UniValue& params, bool fByAccounts)
             obj.push_back(Pair("account",       strAccount));
             obj.push_back(Pair("amount",        ValueFromAmount(nAmount)));
             obj.push_back(Pair("confirmations", (nConf == std::numeric_limits<int>::max() ? 0 : nConf)));
+            if (!fByAccounts)
```
Comment (dcousens) [inline src/wallet/rpcwallet.cpp]: What is `fByAccounts` here?

**Comment by dcousens** [comment]
utACK

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (laanwj) [inline src/wallet/rpcwallet.cpp]: `listreceivedbyaccount`

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -1235,7 +1237,8 @@ UniValue listreceivedbyaddress(const UniValue& params, bool fHelp)
             "    \"address\" : \"receivingaddress\",  (string) The receiving address\n"
             "    \"account\" : \"accountname\",       (string) DEPRECATED. The account of the receiving address. The default account is \"\".\n"
             "    \"amount\" : x.xxx,                  (numeric) The total amount in " + CURRENCY_UNIT + " received by the address\n"
-            "    \"confirmations\" : n                (numeric) The number of confirmations of the most recent transaction included\n"
+            "    \"confirmations\" : n,               (numeric) The number of confirmations of the most recent transaction included\n"
+            "    \"label\" : \"label\"                (string) A comment for the address/transaction, if any\n"
```
Comment (jonasschnelli) [inline src/wallet/rpcwallet.cpp]: Hmm... i think we should not use "transaction" as entity where a label is stored.
What about just using `(string) A comment for the address, if any\n"`?

**Comment by jonasschnelli** [comment]
Tested ACK (fd55571f069640fc2733410061e82b8dd2ed9280) (mind the help text nit).

bildschirmfoto 2015-10-02 um 10 48 47
bildschirmfoto 2015-10-02 um 10 49 04
bildschirmfoto 2015-10-02 um 10 48 51

File: src/wallet/rpcwallet.cpp (hunk 2 above)
Comment (luke-jr) [inline src/wallet/rpcwallet.cpp]: An address and a transaction are essentially the same thing.

Comment (jonasschnelli) [inline src/wallet/rpcwallet.cpp]: > An address and a transaction are essentially the same thing.
^^
Not that I like address re-using: But there are use cases (and users-behaviors) where multiple transactions having outputs to the same address. How could you distinct between these transactions over labels/comments if we would say address==transaction?

We have the term "Address" book where every address has one label. Maybe in future, the Addressbook has identities which could create a payment address over BIP70 or BIP32.

For now i think there are reasons to label/comment an address as well as a transaction.

Comment (luke-jr) [inline src/wallet/rpcwallet.cpp]: There is no need to be distinct across an unsupported use-case. That would just encourage people to do it more often.

**Pull request merged**